src/device.cpp

```diff
@@ -374,6 +374,14 @@ void cvk_device::build_extension_ils_list() {
             MAKE_NAME_VERSION(1, 0, 0, "cl_khr_device_uuid"));
     }
 
+    // Enable cl_khr_fp16 if we have 16-bit storage and shaderFloat16
+    if (is_vulkan_extension_enabled(VK_KHR_16BIT_STORAGE_EXTENSION_NAME) &&
+        is_vulkan_extension_enabled(
+            VK_KHR_SHADER_FLOAT16_INT8_EXTENSION_NAME) &&
```

Comment:
These extensions have become core in 1.2. We need to handle that case as well.

Reply:
Can you clarify what needs to be done here? My impression (maybe incorrect?) was that these extensions would still be advertised even on newer Vulkan versions where they are core. I think the existing code for enabling extension during device initialization also makes this assumption; we only enable 16-bit storage features if the extension is present for example.

Reply:
I could not find a definitive source but I'm pretty sure advertising the extensions is not a requirement. As you say, there are other cases so we might as well deal with all of them with one later change.